src/Features/CSharp/Portable/MakeLocalFunctionStatic/PassVariableExplicitlyInLocalStaticFunctionCodeFixProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(PassVariableExplicitlyInLocalStaticFunctionCodeFixProvider)), Shared]
+    internal sealed class PassVariableExplicitlyInLocalStaticFunctionCodeFixProvider : CodeFixProvider
+    {
+        [ImportingConstructor]
+        public PassVariableExplicitlyInLocalStaticFunctionCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create("CS8421");
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
```

Comment:
batch fixer doesn't work well.  please derive from something like SyntaxEditorBasedCodeFixProvider here.

Reply:
SyntaxEditorBasedCodeFixProvider creates errors with the current code fix and a new issue will be opened to work on the FixAll for this codefix